src/config_listener_mon.erl

```diff
@@ -0,0 +1,84 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(config_listener_mon).
+-behaviour(gen_server).
+-vsn(1).
+
+
+-export([
+    start_link/2
+]).
+
+
+-export([
+    init/1,
+    terminate/2,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    code_change/3
+]).
+
+
+-record(st, {
+    pid,
+    ref
+}).
+
+
+start_link(Module, InitSt) ->
+    proc_lib:start(?MODULE, init, [{self(), Module, InitSt}]).
+
+
+init({Pid, Mod, InitSt}) ->
+    Ref = erlang:monitor(process, Pid),
+    case config_listener:start(Mod, {Mod, Pid}, {Pid, InitSt}) of
```

Comment:
I was wrong it wouldn't break the contract. Since we do pass Pid already 

[Reply = config_listener:start(CallbackModule, {Subscriber, InitialState})](https://github.com/apache/couchdb-config/blob/master/src/config.erl#L264)

The config_listener would make sure Pid is not passed to [Module:handle_config_change](https://github.com/apache/couchdb-config/blob/master/src/config_listener.erl#L43:L44)

```
handle_event({config_change, Sec, Key, Value, Persist}, {Module, {From, State}}) ->
                                                                   |       |
                                                                   x       |
                                                                           V
             case Module:handle_config_change(Sec, Key, Value, Persist, State) of
```